### ISSUE TITLE
[grails 3.1.x ] Adding possibility to define namespace in @Resource

### ISSUE THREAD

**Issue by schmittr** [grails/grails-core#9651]
Hi folks, I have a suggestion:

While writing REST-apis I recently missed the possibility to define the namespace in the @Ressource directive. In my cases this would have saved time, since the automatically generated Controller would have totally sufficed with a given namesapce. But perhaps this normally makes no sense and I'm totally misusing grails.. :smile:

Kind regards

**Comment by jeffbrown** [comment]
@schmittr Can you summarize the behavior that you would expect to result from expressing a namespace for a domain class marked with `@Resource`?

Thanks for the feedback.

**Comment by schmittr** [comment]
Assuming 2 different packages with some classes that are derived from another. e.g.:

```
package org.handsome.foo

@Resource
class A {
	String someField

	static mapping = {
		autoImport false
	    }
}
``` 

and a the same class one
```
package org.handsome.bar

@Resource
class A extends org.handsome.foo.A {
	String someMoreField

	static mapping = {
		autoImport false
	    }
}
```

In order to be able to have two distinct endpoints to those classes, one needs to write two controllers with two different namespaces to get a UrlMapping like:

```
"/bar/a"(controller: "a", namespace: 'bar')
"/foor/a"(controller: "a", namespace: 'foo')
```

But in some cases, the default generated REST-controller would totally suffice and hence in my opinion it would be nice to be able to define the namespace in the domain classes' @Ressource directives.

**Comment by schmittr** [comment]
@jeffbrown what do you think about that feature?
